### PR TITLE
Compile as library and move all json rpc methods to a Connection class

### PR DESCRIPTION
**Pull request by bdhobare**
### Changes:
- Build the library separately as a static library (`libsol`). Split top-level `CMakeLists` into different `CMakeLists` for the examples, the tests and the library itself for easier management.
- Moved all json rpc calls to a `Connection` class that is used as entrypoint to the library and extracted the rpc implementations to `solana.cpp`.
- Moved all WebSocket subscription methods to the namespace `solana::rpc::subscription`
- Refactored the examples to use the `Connection` class.
- Moved all headers to the `include` directory and reorganized methods in the global namespace to the `solana` namespace
```cpp
auto connection = solana::rpc::Connection(config.endpoint);
const auto group = connection.getAccountInfo<mango_v3::MangoGroup>(config.group);
...
const auto market = connection.getAccountInfo<mango_v3::PerpMarket>(perpMarketPk.toBase58());
...
const auto recentBlockhash = connection.getRecentBlockhash();
```
## Tests done:
-  [Transaction](https://explorer.solana.com/tx/5CrFVgyNRSrKmjzJCHEF92XXtgwnykXKy6niui8jyvJnJ2nwi2dJ5EqTYF1rUMqQEYUYcM8kkWNh6fZCGxwbnzNS?cluster=devnet)
-  [PlaceOrder](https://explorer.solana.com/tx/4GMoAJCYLUF6jkooGeg7RXYPUCV7W76SjavahGK3ihFdmPHWd2R3wpnN77wvfGXLKnuN7uSuzjRk62hoa86j5mn4?cluster=devnet)
- [SendTransaction](https://explorer.solana.com/tx/2KN6BVXT1LoNVt7kdvhcy86qvVFFjobRHbCwcEmdtS77CjZfggEHyWX5qavEyiENni3LGFSAHfZcan8ebD2CVpMo?cluster=devnet)
- `./bin/tests`
- Examples

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,20 +9,11 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
...
-target_link_libraries(example-place-order ${CONAN_LIBS})
+# examples
+add_subdirectory(examples)
 
 # tests
-include(CTest)
-add_executable(tests tests/main.cpp)
-target_link_libraries(tests ${CONAN_LIBS})
-add_test(NAME TestMangoGroup COMMAND tests)
+add_subdirectory(tests)
```
Comment (mschneider) [inline CMakeLists.txt:19]: this breaks ctest, test paths are relative to /build

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+include_directories(${solcpp_SOURCE_DIR}/include)
+
```
Comment (mschneider) [inline examples/CMakeLists.txt:2]: use one link_libraries directive in the beginning of the file

File: CMakeLists.txt (hunk 1 above)
Comment (mschneider) [reply CMakeLists.txt:19]: for me ctest is not really important and ci is not using it anyways, so maybe just remove the find & add_test directives

**Comment by mschneider** [comment]
the rest of the codebase is using clang-format with "Visual Studio" style, would be good to keep it in one style or change all files.

File: CMakeLists.txt (hunk 1 above)
Comment (bdhobare) [reply CMakeLists.txt]: Fixed this for `ctest` and `.bin/test` by passing the resources directory directly to the tests.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,215 @@
+---
```
Comment (bdhobare) [inline .clang-format:1]: Added this file for consistency with other contributors. Happy to remove it if it's not necessary.

**Pull request merged**